oac intro pages edits (#104863)
grafana/grafana: docs/sources/observability-as-code/get-started.md

# Get started with Observability as Code

> Grafana provides a suite of tools for **Observability as Code** to help you manage your Grafana resources programmatically and at scale. This approach allows you to define dashboards, data sources, and other configurations in code, enabling version control, automated testing, and reliable deployments through CI/CD pipelines.

Reviewer: ```suggestion
Grafana provides a suite of tools for **Observability as Code** to help you manage your Grafana resources programmatically and at scale. This approach lets you define dashboards, data sources, and other configurations in code, enabling version control, automated testing, and reliable deployments through CI/CD pipelines.
```

> Historically, managing Grafana as code involved various community and Grafana Labs tools, but lacked a single, cohesive story. With Grafana 12, we're introducing foundational improvements, including **new versioned APIs** and official tooling, to provide a clearer path forward.

Reviewer: This sentence is so much better and adds context -- the why someone should care. Do we need to bold ver versioned APIs? Usually, we only bold UI element names. 
```suggestion
Historically, managing Grafana as code involved various community and Grafana Labs tools, but lacked a single, cohesive story. Grafana 12 introduces foundational improvements, including new versioned APIs and official tooling, to provide a clearer path forward.
```

> ## [Grafana CLI (`grafanactl`)](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/grafana-cli)

Reviewer: Moving the docs link to the text so it isn't in the heading. 
```suggestion
## Grafana CLI (`grafanactl`)
```

Use the official command-line tool, `grafanactl`, to interact with your Grafana instances and manage resources via the new APIs.

- It's the recommended tool for automation and direct API interaction, suitable for CI/CD pipelines and local development or ad-hoc tasks. It supports pulling/pushing configs from remote instances, validating configs and more.

> - `grafanactl` works across all environments (Grafana OSS, Enterprise, and Cloud).

Reviewer: We try to minimize parentheses in our writing. 
```suggestion
- `grafanactl` works across all environments for Grafana OSS, Enterprise, and Cloud.
```

> ## [Git Sync](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/intro-git-sync/)

Reviewer: ```suggestion
## Git Sync
```

For an integrated, UI-driven Git workflow focused on dashboards, explore Git Sync.

- Connect folders or entire Grafana instances directly to a GitHub repository to synchronize dashboard definitions, enabling version control, branching, and pull requests directly from Grafana.
- Git Sync offers a simple, out-of-the-box approach for managing dashboards as code.

> - **Note:** Git Sync is an **experimental feature** in Grafana 12, available in Grafana OSS and Enterprise [nightly releases](https://grafana.com/grafana/download/nightly). It is not yet available in Grafana Cloud.

Reviewer: ```suggestion
{{< admonition type="note" >}}
  Git Sync is an **experimental feature** in Grafana 12, available in Grafana OSS and Enterprise [nightly releases](https://grafana.com/grafana/download/nightly). It is not yet available in Grafana Cloud.
  {{< /admonition >}}

Refer to the [Git Sync documentation](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/observability-as-code/provision-resources/intro-git-sync/) to learn more.

```

> ## [Direct API Usage](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/developers/http_api/apis/)

Reviewer: ```suggestion
## Direct API usage
```

For maximum flexibility, advanced use cases, or building custom tooling, you can interact directly with the underlying versioned APIs.

- This approach requires handling HTTP requests and responses but provides complete control over resource management.
- `grafanactl`, Git Sync, and the Foundation SDK are all built on top of these APIs.

> - To understand Dashboard Schemas accepted by the APIs, look at the [json models documentation](json-models/)

Reviewer: ```suggestion
- To understand Dashboard Schemas accepted by the APIs, refer to the [JSON models documentation](json-models/).

Refer to the [Grafana APIs](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/developers/http_api/apis/) documentation for more information.
```

> ## [Foundation SDK](foundation-sdk)

Reviewer: ```suggestion
## Foundation SDK
```

To programmatically define your Grafana resources (like dashboards or alerts) using familiar programming languages, use Foundation SDK.

> - Define resources using strongly-typed builders in languages like Go, TypeScript, Python, Java, and PHP.

Reviewer: We don't use hyphens after verbs ending in -ly.
```suggestion
- Define resources using strongly typed builders in languages like Go, TypeScript, Python, Java, and PHP.
```

- Avoid crafting complex JSON manually and integrate resource generation into your existing development workflows.
- Catch errors at compile time and easily integrate resource generation into your CI/CD pipelines.
- Use in conjunction with `grafanactl` to push your programmatically generated resources.

> 

Reviewer: ```suggestion

Refer to the [Foundation SDK](./foundation-sdk) documentation for more information.

```

## Additional Observability as Code tools

> If you're already using established Infrastructure as Code or other configuration management tools, Grafana offers integrations to manage resources within your existing workflows:

Reviewer: ```suggestion
If you're already using established Infrastructure as Code or other configuration management tools, Grafana offers integrations to manage resources within your existing workflows.
```
